pkgs/development/python-modules/dns-lexicon/default.nix

```diff
@@ -0,0 +1,42 @@
+{ buildPythonPackage
+, fetchPypi
+, poetry-core
+, beautifulsoup4
+, cryptography
+, importlib-metadata
+, pyyaml
+, requests
+, tldextract
+, lib
+}:
+
+buildPythonPackage rec {
+  pname = "dns_lexicon";
+  version = "3.14.1";
+  format = "pyproject";
+
+  nativeBuildInputs = [ poetry-core ];
+
+  propagatedBuildInputs = [
+    beautifulsoup4
+    cryptography
+    importlib-metadata
+    pyyaml
+    requests
+    tldextract
+  ];
+
```

Comment:
How about adding the passthru.optional-dependencies?
https://github.com/AnalogJ/lexicon/blob/f06c69039a4e08cc576817e214d014307922969f/pyproject.toml#L54

https://github.com/NixOS/nixpkgs/blob/master/doc/languages-frameworks/python.section.md#how-to-add-optional-dependencies-python-optional-dependencies

Reply:
Can you check what I added?